src/v/s3/client.cc

```diff
@@ -0,0 +1,427 @@
+#include "s3/client.h"
+
+#include "bytes/iobuf.h"
+#include "bytes/iobuf_istreambuf.h"
+#include "hashing/secure.h"
+#include "s3/error.h"
+#include "s3/logger.h"
+#include "s3/signature.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/iostream.hh>
+#include <seastar/core/temporary_buffer.hh>
+#include <seastar/net/dns.hh>
+#include <seastar/net/inet_address.hh>
+#include <seastar/net/tls.hh>
+
+#include <boost/beast/core/error.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/xml_parser.hpp>
+#include <fmt/core.h>
+#include <gnutls/crypto.h>
+
+namespace s3 {
+
+struct aws_header_names {
+    static constexpr boost::beast::string_view prefix = "prefix";
```

Comment:
minor cosmetic nit: i bet in the future, boost::string_view will use `std::string_view`. when we update the boost version, we should update this code too. so maybe write a todo comment.

Reply:
makes sense.